git-user-count/user-count.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+set -e
+
+basedir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# input
+repositories="$basedir/repositories.txt"
+
+# outputs
+userListFile="$basedir/gradle-enterprise-users.txt"
+userUniqueListFile="$basedir/gradle-enterprise-unique-users.txt"
+usersByRepoFile="$basedir/gradle-enterprise-users-by-repo.csv"
+checkout_area=.tmp/repos
+
+# options
+since=30
+shallow_clone=true
+git_options=
+
+yellow='\033[1;33m'
+nc='\033[0m'
+
+# -b option lets you specify a branch to checkout, store it in a variable
+# -s option lets you specify a number of days to count commits from, store it in a variable otherwise default to 30
+# -c option disables shallow cloning
+# -o option lets you specify aditional git cloning options
+while getopts b:s:co: option
+do
+  case "${option}" in
+    b) branch=${OPTARG};;
+    s) since=${OPTARG};;
+    c) shallow_clone=false;;
+    o) git_options=${OPTARG};;
+    *) echo "Usage: $0 [-b branch_name] [-s since_days] [-o git_options] [-c]" >&2
+       exit 1;;
+  esac
+done
+
+# Check that both -s and -c cannot be specified at the same time
+if [ "$shallow_clone" = false ] && [ -n "$since" ]; then
+  echo "Cannot specify both -s and -c options at the same time" >&2
+  exit 1
```

Comment:
I suggest printing the usage here as well. Maybe extract printing the usage to a function.